src/utils/postgre.ts

```diff
@@ -0,0 +1,56 @@
+import { identity, pipe } from "fp-ts/lib/function";
+import { DatabaseError, Pool, QueryResult } from "pg";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { getRequiredStringEnv } from "@pagopa/io-functions-commons/dist/src/utils/env";
+import { NumberFromString } from "@pagopa/ts-commons/lib/numbers";
+import { cli } from "cli-ux";
+import chalk from "chalk";
+
+// eslint-disable-next-line functional/no-let
+let singletonPool: Pool;
+export const getPool = (): Pool => {
+  if (!singletonPool) {
+    singletonPool = new Pool({
+      database: getRequiredStringEnv("POSTGRE_DB_NAME"),
+      host: getRequiredStringEnv("POSTGRE_DB_HOST"),
+      idleTimeoutMillis: pipe(
+        parseInt(getRequiredStringEnv("POSTGRE_DB_IDLE_TIMEOUT")),
+        NumberFromString.decode,
+        E.foldW(() => 3000, identity)
+      ),
+      max: 20,
+      password: getRequiredStringEnv("POSTGRE_DB_PASSWORD"),
+      port: pipe(
+        getRequiredStringEnv("POSTGRE_DB_PORT"),
+        NumberFromString.decode,
+        E.foldW(() => 5432, identity)
+      ),
+      ssl: true,
+      user: getRequiredStringEnv("POSTGRE_DB_USER"),
+    });
+  }
+  return singletonPool;
```

Comment:
refactor in a functional way by wrapping `sigletonPool` into an `Option` with `fromNullable`

Reply:
Check it 075832e maybe it's ok now.